src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs

```diff
@@ -175,6 +180,66 @@ public ZipArchive(Stream stream, ZipArchiveMode mode, bool leaveOpen, Encoding?
             }
         }
 
+
+        public ZipArchive(Stream stream, ZipArchiveMode mode, bool leaveOpen, Encoding? entryNameEncoding, string? defaultPassword, ZipArchiveEntry.EncryptionMethod defaultEncryption)
+            : this(mode, leaveOpen, entryNameEncoding, backingStream: null, archiveStream: DecideArchiveStream(mode, stream), defaultPassword: defaultPassword, defaultEncryption: defaultEncryption)
+        {
+            ArgumentNullException.ThrowIfNull(stream);
+
+            Stream? extraTempStream = null;
+
+            try
+            {
+                _backingStream = null;
+
+                if (ValidateMode(mode, stream))
+                {
+                    _backingStream = stream;
+                    extraTempStream = stream = new MemoryStream();
+                    _backingStream.CopyTo(stream);
+                    stream.Seek(0, SeekOrigin.Begin);
+                }
+
+                _archiveStream = DecideArchiveStream(mode, stream);
+
+                switch (mode)
```

Comment:
why is this custom logic needed? I would expect all the initialization to be done by one of the other constructors we are calling via `this(....)`